QueueUsingStack2.java

```diff
@@ -0,0 +1,74 @@
+import java.util.Stack;
+
+public class QueueUsingStack2 {
+	// This solution referred from Leetcode 
+	
+	// In this method move elements from reference stack to queue stack when needed,
+	// i.e., when we need to peek/pop but the output stack is empty.
+	
+	Stack<Integer> stack1; // reference stack
+    Stack<Integer> stack2; // queue stack
+    
+    /** Initialize your data structure here. */
+    public QueueUsingStack2() {
+       stack1 = new Stack<>();
+        stack2 = new Stack<>();
+    }
+    
+    /** Push element x to the back of queue. */
+    public void push(int x) {
+          stack1.push(x);
+    }
+    
+    /** Removes the element from in front of queue and returns that element. */
+    // first we update the queue stack with all the newly pushed elements
+    // and then pop the element from queue stack 
```

Comment:
Would you like to comment on time and space complexity for all operations in both the approaches.

Reply:
I am not sure as to what exactly can be the complexity, but it seems to be better solution because we only move elements from a stack, when needed i.e. when a peek function is called. I am confused whether it's still O(1) or O(n).